### PR TITLE
Forced gender fixes

### PR DESCRIPTION
**Pull request by Mqiib**
Makes forced genders not get circumvented by all the standard methods I could think of (magic mirror, cloning, and xenobio gender potion). This only applies to polysmorphs currently.

#### Changelog

:cl:  
bugfix: Forced gender harder to get around for species that have it.
/:cl:

### CODE REVIEW

**Comment by ghost** [comment]
no more xeno femboy benis

File: code/modules/research/xenobiology/xenobiology.dm [hunk 1]
```diff
@@ -933,14 +933,26 @@
 	if(L.gender != MALE && L.gender != FEMALE)
 		to_chat(user, "<span class='warning'>The potion can only be used on gendered things!</span>")
 		return
-
-	if(L.gender == MALE)
-		L.gender = FEMALE
-		L.visible_message("<span class='boldnotice'>[L] suddenly looks more feminine!</span>", "<span class='boldwarning'>You suddenly feel more feminine!</span>")
+	if(iscarbon(L))
+		var/mob/living/carbon/C = L
+		if(!(MGENDER in C.dna.species.species_traits) && !(FGENDER in C.dna.species.species_traits) && !(AGENDER in C.dna.species.species_traits))
```
Comment (jcatjmeow) [inline code/modules/research/xenobiology/xenobiology.dm]:
```suggestion
		if(C.dna && !(MGENDER in C.dna.species.species_traits) && !(FGENDER in C.dna.species.species_traits) && !(AGENDER in C.dna.species.species_traits))
```

**Review by jcatjmeow — CHANGES_REQUESTED** [review]
Add this so that if the thing is a carbon but has no dna (for whatever fucky reason) it wont runtime.

File: code/modules/research/xenobiology/xenobiology.dm [hunk 2]
```diff
@@ -933,14 +933,26 @@
 	if(L.gender != MALE && L.gender != FEMALE)
 		to_chat(user, "<span class='warning'>The potion can only be used on gendered things!</span>")
 		return
-
-	if(L.gender == MALE)
-		L.gender = FEMALE
-		L.visible_message("<span class='boldnotice'>[L] suddenly looks more feminine!</span>", "<span class='boldwarning'>You suddenly feel more feminine!</span>")
+	if(iscarbon(L))
+		var/mob/living/carbon/C = L
+		if(C.dna && !(MGENDER in C.dna.species.species_traits) && !(FGENDER in C.dna.species.species_traits) && !(AGENDER in C.dna.species.species_traits)))
```
Comment (jcatjmeow) [inline code/modules/research/xenobiology/xenobiology.dm]:
```suggestion
		if(C.dna && !(MGENDER in C.dna.species.species_traits) && !(FGENDER in C.dna.species.species_traits) && !(AGENDER in C.dna.species.species_traits))
```

**Review by jcatjmeow — CHANGES_REQUESTED** [review]
sorry typo

File: code/modules/research/xenobiology/xenobiology.dm (hunk 2 above)
Comment (Mqiib) [reply code/modules/research/xenobiology/xenobiology.dm]: I TRUSTED YOU

**Comment by athena413** [comment]
ETHEREALS DONT HAVE GENDER PLEASE FIX IT FOR ETHEREALS TOO

**Comment by Mqiib** [comment]
Etherials don't have the same issues polysmorphs have. Them not having a gender is already accounted for, as far as I know. If you can tell me exactly what isn't working right about them I could look into it.

**Comment by Mqiib** [comment]
Ah don't merge this yet I have to fix the gender reassignment surgery too

**Review by SomeguyManperson — APPROVED** [review]

**Pull request merged**